plugin/pkg/shoot/resourcereservation/apis/shootresourcereservation/types.go

```diff
@@ -16,4 +16,6 @@ type Configuration struct {
 	// UseGKEFormula enables the calculation of resource reservations based on
 	// the CPU and memory resources available for a machine type.
 	UseGKEFormula bool `json:"useGKEFormula"`
+	// LabelSelector optionally defines a label selector for which the GKE formula should be applied
+	LabelSelector string `json:"labelSelector,omitempty"`
```

Comment:
```suggestion
	Selector *metav1.LabelSelector
```

We can remove all `json` tags in the internal API definition.